03-explanation/01-foundations/automation-paradigms/overview.mdx

```diff
@@ -0,0 +1,107 @@
+---
+title: "Automation Paradigms"
+description: "Understand AskUI's single-step commands and agentic instruction approaches"
+---
+
+AskUI operates through two distinct paradigms that reflect different approaches to automation complexity. Understanding these paradigms helps explain why AskUI can handle both fine-grained controlled tasks and complex, adaptive workflows.
+
+## Two Automation Paradigms
+
+### Agentic Instructions: Intent-Based Automation
+
+Agentic instructions let you describe what you want to accomplish in natural language. The agent interprets your goals and determines the necessary steps independently.
+
+```python
+# Agentic approach: intent-based control
+with VisionAgent() as agent:
+    agent.act("Fill out the login form with username john.doe and submit it")
+    agent.act("Open the settings menu and navigate to display preferences")
+```
+
+**Recommended for most use cases**:
+- **Natural language** describes what you want to accomplish
+- **Adaptive behavior** handles interface changes automatically
+- **Complex workflows** with multiple steps
+- **Reduced maintenance** as interfaces evolve
+
+<Tip>**Recommendation**: Use agentic instructions as your primary approach. Single-step commands serve as an escape hatch for cases where agentic instructions aren't reliable enough yet.</Tip>
+
+### Single-Step Commands: Fine-Grained Control
+
+Single-step commands represent the traditional automation approach where each interaction is explicitly defined and controlled. While AI models are still involved in understanding visual elements and executing commands, this paradigm provides explicit control over the automation workflow.
+
+```python
+# Single-step approach: explicit control
+with VisionAgent() as agent:
+    agent.mouse_scroll(0, 10)
+    agent.type("john.doe")
+    agent.click("Submit button")
+```
+
+Use this approach when:
+- **Agentic instructions fail** to handle specific cases reliably
+- **Fine-grained control** is absolutely necessary
+- **Debugging** specific interaction failures
+- **Legacy integrations** require explicit control
+
+## AI Models in Both Paradigms
```

Comment:
We should introduce here LocatorModels, GetModels and ActModels. See https://github.com/askui/vision-agent?tab=readme-ov-file#-custom-models

**LocatorModels** (NL, Screenshot -> Model -> Point)
- Find Text
- Find Images / Icon
- Find Element Types (textfields)
- Find Element Descriptions (Prompts) 

**GetModels ** (NL, Image(, Response Schema -> Model -> Structued Output)
- Answer Yes/No questions
- ...

**ActModels** (Goal, Tools -> Model -> Sequesnce Of Actions)
- Autonoumes behaveing

Reply:
I added model types here and linked to https://github.com/askui/vision-agent/blob/main/src/askui/models/models.py